MvvmCross/Platforms/Uap/Views/MvxApplication.cs

```diff
@@ -118,28 +117,18 @@ protected virtual void OnNavigationFailed(object sender, NavigationFailedEventAr
             throw new Exception("Failed to load Page " + e.SourcePageType.FullName);
         }
 
-        private async void OnSuspending(object sender, SuspendingEventArgs e)
+        private async void OnEnteredBackground(object sender, object e)
```

Comment:
This isn't sufficient - the point of deferrals in the OnSuspending is that if there is substantial amount of work to be done the app gets a chance to finish it before the app is suspended. I agree that we should move the Suspend logic to entering background but we need to make sure that if the app gets suspended whilst the Suspend logic is being run, we capture the deferral and only release it once the Suspend logic is complete.
Also, we can't just ignore the Resume logic - what happens if the app does in fact get suspended. As we just going to ignore resuming the application?

Reply:
@nickrandolph  Thanks for the feedback. I didn't realize you could get the deferral off the EnteredBackgroundEventArgs. Silly of me to leave it out.

As for the Resuming handler, do we actually have to resume the suspension? Per what @MartinZikmund wrote in the original issue:

>This means when the user switches back to the minimized app, it first restores the state. But that >is not necessary at all, because the full app state is still in memory and there is no additional >work required. Basically, suspension can be seen as a precautionary measure in case the system >runs out of resources and needs to kill the app to free some up.
>
>When the system kills a suspended app, when the user launches it or switches back to it, it will >now go through the full launch lifetime

Calling "Resume(suspension)" when the app was suspended but not terminated is what causes the original issue. As Martin mentioned, in the case of termination, we could call resume from OnLaunch/OnActivated. His opinion on this was:

>the better solution may just be to leave this up to the developer to decide if she wants to restore >the state saved in suspension or not. This would give better flexibility to the developers using >MvvmCross and would also preserve the current behavior which is that state is not restored.

At the moment, I'm not sure which I prefer. But what I have changed will allow the app to be suspended and resumed as long as it is not terminated by windows. If it is, it will not resume from the suspension point, which is currently does not since it is broken. Developers could handle the relevant events (Launch/Activated) themselves if they want to resume from suspension after termination.